### PR TITLE
fix potential mem leakage in nodeorder.go

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

`WithCancle` context should GC  when Context Complete to release the resources.

### CODE REVIEW

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 1]
```diff
@@ -213,7 +214,6 @@ func (pp *nodeOrderPlugin) OnSessionOpen(ssn *framework.Session) {
 			Resources: []config.ResourceSpec{{Name: "cpu", Weight: 1}, {Name: "memory", Weight: 1}},
 		},
 	}
-	noderesources.NewFit(mostAllocatedArgs, handle, fts)
```
Comment (kerthcet) [inline pkg/scheduler/plugins/nodeorder/nodeorder.go:216]: I think this is redundant, correct me if I missed something.

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by kerthcet** [comment]
This is approved but it suggests need another review. cc @william-wang PTAL

File: pkg/scheduler/plugins/nodeorder/nodeorder.go [hunk 2]
```diff
@@ -368,6 +368,8 @@ func interPodAffinityScore(
 	workerNum := 16
 	errCh := make(chan error, workerNum)
 	parallelizeContext, parallelizeCancel := context.WithCancel(context.TODO())
+	defer parallelizeCancel()
+
 	workqueue.ParallelizeUntil(parallelizeContext, workerNum, len(nodes), func(index int) {
```
[the remark this answers is not in the document]
Comment (kerthcet) [reply pkg/scheduler/plugins/nodeorder/nodeorder.go:373]: Nope, I think this is intended, so when we got a unsuccessful status in L378, we'll cancel all the forked context by calling `parallelizeCancel`.

[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pkg/scheduler/plugins/nodeorder/nodeorder.go:373]: Good, i missed it

**Review by Thor-wl — APPROVED** [review]

**Pull request merged**